test/iterTest.mo

```diff
@@ -59,6 +59,24 @@ do {
   };
 };
 
+do {
+  Debug.print("  filter");
+
+  let isOdd = func (x : Int) : Bool {
+    x % 2 == 1;
+  };
+
+  let _actual = Iter.filter<Nat>([ 1, 2, 3 ].vals(), isOdd);
+  let actual = [var 0, 0];
+  Iter.iterate<Nat>(_actual, func (x, i) { actual[i] := x; });
+
+  let expected = [1, 3];
+
+  for (i in actual.keys()) {
+    assert(actual[i] == expected[i]);
```

Comment:
Can't we just write `assert(actual == expected);`, i.e. without the surrounding loop?

Reply:
It's actually better, because else it wouldn't notice if `actual` is too short